### PR TITLE
Feature/stats

### PR DESCRIPTION
**Pull request by AshakaE**
# Statistics of wikis created/edited and by user

_Default date args are 7 days apart from now, date args are unixtime_

## How should this be tested?

1. wikisCreated image
2. wikisEdited image
3. wikisCreatedByUser image
4. wikisEditedByUser image

## Notes or observations

_I'm not sure how to use the interval hours in the query for wikisCreated/Edited 🤔_

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/558
cc @Softdev1

### CODE REVIEW

File: src/App/stats.resolver.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  @Field(() => String)
+  amount!: string
+}
+@ObjectType()
+export class WikiUserStats {
+  @Field()
+  address!: string
+
+  @Field(() => String)
+  amount!: string
```
Comment (kesar) [inline src/App/stats.resolver.ts:32]: amount string 😢 ?

sad

**Comment by kesar** [comment]
> I'm not sure how to use the interval hours in the query for wikisCreated/Edited 🤔

for example, if I use startOn (1 day ago) endOn (now) interval 1h, you should return an array of wikistats perhour.

[ { startOn: 1 day ago, endOn: 1 day ago -1h, amount: 4}, { startOn: 1 day ago - 1h, endOn: 1 day ago -2h, amount: 2},... ]


its also missing the graph query to get a leaderboard:

"
number of wikis created
number of wikis edited
filtered by startdate and enddate, with interval in hours
it should return starton, endon, amount

number of wikis created by users. it should return (address, amount)
number of wikis edited by users. it should return (address, amount)
filtered by startdate and enddate" <- by users not by user (but its fine, you can keep that one too)

the main point its to get a leaderboard of users that have created more wikis in last week.

startOn <- 1 week ago
endOn <- today

expecting

[ { address: 0x0, amount: 4}, { address: 0x1, amount: 1} ]

**Comment by AshakaE** [comment]
> > I'm not sure how to use the interval hours in the query for wikisCreated/Edited 🤔
> 
> for example, if I use startOn (1 day ago) endOn (now) interval 1h, you should return an array of wikistats perhour.
> 
> [ { startOn: 1 day ago, endOn: 1 day ago -1h, amount: 4}, { startOn: 1 day ago - 1h, endOn: 1 day ago -2h, amount: 2},... ]
> 
> its also missing the graph query to get a leaderboard:
> 
> " number of wikis created number of wikis edited filtered by startdate and enddate, with interval in hours it should return starton, endon, amount
> 
> number of wikis created by users. it should return (address, amount) number of wikis edited by users. it should return (address, amount) filtered by startdate and enddate" <- by users not by user (but its fine, you can keep that one too)
> 
> the main point its to get a leaderboard of users that have created more wikis in last week.
> 
> startOn <- 1 week ago endOn <- today
> 
> expecting
> 
> [ { address: 0x0, amount: 4}, { address: 0x1, amount: 1} ]

Ooh, It's clear now

File: src/App/stats.resolver.ts (hunk 1 above)
Comment (AshakaE) [reply src/App/stats.resolver.ts:32]: image

**Review by Softdev1 — APPROVED** [review]

**Pull request merged**